Gotrue/Client.cs

```diff
@@ -194,27 +84,25 @@ public static void Initialize(ClientOptions options = null, Action<Client> callb
         /// </summary>
         /// <param name="options"></param>
         /// <returns></returns>
-        public static async Task<Client> InitializeAsync(ClientOptions options = null)
+        public async Task<IGotrueClient<User, Session>> InitializeAsync(ClientOptions options = null)
         {
-            instance = new Client();
-
             if (options == null)
                 options = new ClientOptions();
 
-            instance.Options = options;
-            instance.AutoRefreshToken = options.AutoRefreshToken;
-            instance.ShouldPersistSession = options.PersistSession;
-            instance.SessionPersistor = options.SessionPersistor;
-            instance.SessionRetriever = options.SessionRetriever;
-            instance.SessionDestroyer = options.SessionDestroyer;
+            Options = options;
+            AutoRefreshToken = options.AutoRefreshToken;
+            ShouldPersistSession = options.PersistSession;
+            SessionPersistor = options.SessionPersistor;
+            SessionRetriever = options.SessionRetriever;
+            SessionDestroyer = options.SessionDestroyer;
 
-            instance.api = new Api(options.Url, options.Headers);
+            api = new Api(options.Url, options.Headers);
 
             // Retrieve the session
-            if (instance.ShouldPersistSession)
-                await instance.RetrieveSessionAsync();
+            if (ShouldPersistSession)
+                await RetrieveSessionAsync();
 
-            return instance;
+            return this;
         }
```

Comment:
Could probably be moved to a constructor.
eg:
```csharp
public Client (IGotrueApi<User, Session> api, ClientOptions options) 
{
            Options = options;
            AutoRefreshToken = options.AutoRefreshToken;
            ShouldPersistSession = options.PersistSession;
            SessionPersistor = options.SessionPersistor;
            SessionRetriever = options.SessionRetriever;
            SessionDestroyer = options.SessionDestroyer;

            this.api = api;

            // Retrieve the session
            if (ShouldPersistSession)
                await RetrieveSessionAsync();
}

public async Task<IGotrueClient<User, Session>> InitializeAsync (ClientOptions options = null) 
{
            if (options == null)
                options = new ClientOptions();

            return new Client(new Api(options.Url, options.Headers), options);
}
```

Reply:
Hm. I've also wanted to move this to a constructor, but it needs that async call to `RetrieveSessionAsync()` - though I suppose we can separate that out of the constructor as a change.

Reply:
oops did not see that await! You're right though